x-pack/test/functional/apps/dashboard/dashboard_lens_by_value.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common', 'dashboard', 'visualize', 'lens']);
+
+  const log = getService('log');
+  const find = getService('find');
+  const esArchiver = getService('esArchiver');
+  const dashboardPanelActions = getService('dashboardPanelActions');
+  const dashboardVisualizations = getService('dashboardVisualizations');
+
+  async function createAndAddLensByValue() {
```

Comment:
this function might be helpful to throw into a page objects somewhere for re-use

Reply:
Good call, I've added this to the lens page, and made it generic, to be used either by value or by reference.